lib/sc_client.rb

```diff
@@ -176,9 +186,12 @@ def connect
       end
 
       @ws.onclose do
-        @on_disconnected.call if @on_disconnected
-        if @enable_reconnection
-          reconnect
+        if should_reconnect
+          @reconnect_interval = @max_reconnect_interval if @reconnect_interval > @max_reconnect_interval
```

Comment:
why remove 'reconnect' method, there is not much change in the re connection logic and separate method provides us re usability in future

Reply:
If we call connect method from reconnect, the signature of websocket instance variables keep on changing and hence onopen cannot be called, so added it in onclose method of websocket.